src/main/java/apoc/map/Maps.java

```diff
@@ -178,18 +178,47 @@ public Object get(@Name("map") Map<String,Object> map, @Name("key") String key,
     }
 
     @UserFunction
-    @Description("apoc.map.removeKey(map,key)")
-    public Map<String,Object> removeKey(@Name("map") Map<String,Object> map, @Name("key") String key) {
+    @Description("apoc.map.removeKey(map,key,{recursive:true/false}) - remove the key from the map (recursively if recursive is true)")
+    public Map<String,Object> removeKey(@Name("map") Map<String,Object> map, @Name("key") String key,  @Name(value="config", defaultValue = "{}") Map<String, Object> config) {
+        if (!map.containsKey(key)) {
+            return map;
+        }
         Map<String, Object> res = new LinkedHashMap<>(map);
         res.remove(key);
+        if (config!=null && config.get("recursive")==Boolean.TRUE) {
+            for (Map.Entry<String, Object> entry : res.entrySet()) {
+                if (entry.getValue() instanceof Map) {
+                    Map<String, Object> updatedMap = removeKey((Map<String, Object>) entry.getValue(), key, config);
+                    if (updatedMap!=entry.getValue()) {
+                        entry.setValue(updatedMap);
+                    }
+                } else if (entry.getValue() instanceof Collection) {
+                    Collection<Map<String, Object>> values = (Collection<Map<String, Object>>) entry.getValue();
+                    List<Map<String, Object>> updatedValues = values.stream().map(value -> removeKey(value, key, config)).collect(Collectors.toList());
+                    entry.setValue(updatedValues);
+                }
+            }
+        }
         return res;
     }
 
     @UserFunction
-    @Description("apoc.map.removeKeys(map,keys)")
-    public Map<String,Object> removeKeys(@Name("map") Map<String,Object> map, @Name("keys") List<String> keys) {
+    @Description("apoc.map.removeKeys(map,[keys],{recursive:true/false}) - remove the keys from the map (recursively if recursive is true)")
+    public Map<String,Object> removeKeys(@Name("map") Map<String,Object> map, @Name("keys") List<String> keys, @Name(value="config", defaultValue = "{}") Map<String, Object> config) {
         Map<String, Object> res = new LinkedHashMap<>(map);
         res.keySet().removeAll(keys);
+        if (config!=null && config.get("recursive")==Boolean.TRUE) {
```

Comment:
can you add a null check over the config?
```suggestion
        if (config!=null) {
             config = Collections.emptyMap();
        }
```

And then:
```
if (Util.toBoolean(config.getOrDefault("recursive", false))) {
....
}
```

Reply:
`if (config == null) {`

Reply:
ok